### PR TITLE
Simplify transport modes

### PR DESCRIPTION
**Pull request by christophfink**
This is a first step for implementing a different DetailedItinerariesComputer. It’s a separate PR to ease running the tests.

### CODE REVIEW

**Comment by christophfink** [comment]
I created a separate issue for updating the respective documentation, see #290

**Comment by HTenkanen** [comment]
@christophfink Is this ready for review? I see lots of things happening so just wanted to make sure before diving in? 🙂

File: tests/test_transport_mode.py
Comment (HTenkanen) [inline tests/test_transport_mode.py:1]: Should this one also test the case if user passes incorrect TransportMode? Like typos (CRA) or modes that are not supported (helicopter).

**Review by HTenkanen — APPROVED** [review]
One small thing to consider in tests, but otherwise looks good to me! I like this logic of reducing a bit the complexity of r5 from the user 👍🏻

File: tests/test_transport_mode.py
Comment (christophfink) [reply tests/test_transport_mode.py:1]: Good point! I added such tests. Beyond that, I also made sure the case of a transport mode passed in as `str` does not make a difference, and added tests to check that that works.

**Comment by christophfink** [comment]
@HTenkanen : just pushed another small commit to address the concern you brought up, could you take another look?

**Pull request merged**